Mage.Sets/src/mage/cards/r/RazorfieldRipper.java

```diff
@@ -0,0 +1,139 @@
+import mage.abilities.effects.common.counter.GetEnergyCountersControllerEffect;
+import mage.constants.*;
+import mage.abilities.keyword.ReconfigureAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class RazorfieldRipper extends CardImpl {
+
+    private static final DynamicValue xValue = new CountersSourcePlayerCount(CounterType.ENERGY);
+
+    public RazorfieldRipper(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{2}{W}");
+        
+        this.subtype.add(SubType.EQUIPMENT);
+        this.subtype.add(SubType.RHINO);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever Razorfield Ripper or equipped creature attacks, you get {E}, then it gets +X/+X until end of turn, where X is the amount of {E} you have.
+        Ability ability = new RazorfieldRipperTriggeredAbility(new GetEnergyCountersControllerEffect(1));
+        ability.addEffect(new BoostTargetEffect(xValue, xValue, Duration.EndOfTurn));
+        this.addAbility(ability);
+
+        // Reconfigure--Pay {2} or {E}{E}{E}.
+        Cost cost = new OrCost("Pay {2} or {E}{E}{E}", new GenericManaCost(2), new PayEnergyCost(3));
+        this.addAbility(new ReconfigureAbility(cost));
+
+    }
+
+    private RazorfieldRipper(final RazorfieldRipper card) {
+        super(card);
+    }
+
+    @Override
+    public RazorfieldRipper copy() {
+        return new RazorfieldRipper(this);
+    }
+}
+
+class RazorfieldRipperEffect extends OneShotEffect {
+
+    RazorfieldRipperEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private RazorfieldRipperEffect(final RazorfieldRipperEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RazorfieldRipperEffect copy() {
+        return new RazorfieldRipperEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        game.informPlayers("Target: " + getTargetPointer().toString());
+        int xValue = player.getCountersCount(CounterType.ENERGY);
+        Effect boostEffect  = new BoostTargetEffect(xValue, 3, Duration.EndOfTurn);
+        boostEffect.setTargetPointer(getTargetPointer());
+        boostEffect.apply(game, source);
+        return true;
+    }
+}
+
+class RazorfieldRipperTriggeredAbility extends TriggeredAbilityImpl {
+
+    RazorfieldRipperTriggeredAbility(Effect effect) {
+        super(Zone.BATTLEFIELD, effect);
+        setTriggerPhrase("Whenever {this} or equipped creature attacks");
```

Comment:
need to end with comma and trailing space

Reply:
fixed